packages/ts-client/src/messageQueue.ts

```diff
@@ -0,0 +1,34 @@
+type MessageQueueDeps = {
+  checkIsReconnecting: () => boolean;
+  sendMessage: (message: Uint8Array) => void;
+};
+
+export class MessageQueue {
+  private queuedMessages: Uint8Array[] = [];
+  private checkIsReconnecting: MessageQueueDeps['checkIsReconnecting'];
+  private sendMessage: MessageQueueDeps['sendMessage'];
+
+  constructor(deps: MessageQueueDeps) {
```

Comment:
nit: I have different association with word `deps`, than function params (so I would rather call it `params`?) Or maybe just:
```suggestion
  constructor({checkIsReconnecting, sendMessage} : MessageQueueParams) {
```

Reply:
done, what's the association though? i meant dependencies